internal/code-generator/generator/samplesfreq.go

```diff
@@ -93,22 +92,23 @@ func getFrequencies(samplesDir string) (*samplesFrequencies, error) {
 	}, nil
 }
 
-func getSamples(samplesDir string, langDir os.FileInfo) ([]string, error) {
-	const samplesSubDir = "filenames"
-	samples := []string{}
-	path := filepath.Join(samplesDir, langDir.Name())
-	entries, err := ioutil.ReadDir(path)
+func getSamplesFrom(samplesLangDir string) ([]string, error) {
+	const samplesLangFilesDir = "filenames"
+	var samples []string
```

Comment:
Please move this down to just before the loop, since it's not needed before then.